src/groups/bmq/bmqimp/bmqimp_application.cpp

```diff
@@ -440,21 +438,10 @@ bmqt::GenericResult::Enum Application::startChannel()
     BSLS_ASSERT_SAFE(d_brokerSession.state() ==
                      bmqimp::BrokerSession::State::e_STARTING);
 
-    int rc = d_channelFactoryPipeline.start();
-    if (rc != 0) {
-        BALL_LOG_ERROR << "Failed to start channelFactoryPipeline [rc: " << rc
-                       << "]";
-        return bmqt::GenericResult::e_UNKNOWN;  // RETURN
-    }
-
-    bdlb::ScopeExitAny pipelineScopeGuard(
-        bdlf::BindUtil::bind(&ChannelFactoryPipeline::stop,
-                             &d_channelFactoryPipeline));
-
-    // Connect to the broker.
+    // 1. Prepare and validate connection parameters
     bmqio::TCPEndpoint endpoint(d_sessionOptions.brokerUri());
```

Comment:
Reordered to check `brokerUri` first, this check is fast and cheap.
If we are going to fail, we can do it before starting the channel factory pipeline